packages/api/src/command/medical/patient/reset-external-data.ts

```diff
@@ -0,0 +1,42 @@
+import { MedicalDataSource, isMedicalDataSource } from "@metriport/core/external/index";
+import { BadRequestError } from "@metriport/shared";
+import { PatientModel } from "../../../models/medical/patient";
+import { getPatientModelOrFail } from "./get-patient";
+import { executeOnDBTx } from "../../../models/transaction-wrapper";
+
+export type ResetExternalData = {
+  cxId: string;
+  patientId: string;
+  source: string;
+};
+
+export const resetExternalData = async ({
+  cxId,
+  patientId,
+  source,
+}: ResetExternalData): Promise<void> => {
+  if (!isMedicalDataSource(source)) {
+    throw new BadRequestError(
+      `Invalid source. Must be one of: ${Object.values(MedicalDataSource).join(", ")}`
+    );
+  }
+
+  await executeOnDBTx(PatientModel.prototype, async transaction => {
+    const patient = await getPatientModelOrFail({ cxId, id: patientId, transaction, lock: true });
+
+    const updatedPatient = await patient.update(
+      {
+        data: {
+          ...patient.data,
+          externalData: {
+            ...patient.data.externalData,
+            [source]: undefined,
```

Comment:
nit, not sure if normally an earlier state is having the source undefined, or not in the object at all. this will leave your source with a value of undefined in the DB.
```
import { omit } from lodash;
// stuff
...omit(patient.data.externalData, source)
```

Reply:
I tested this locally and this will remove the key value from the object and from the db ie. it will not show undefined

Reply:
sweet